mlir/lib/Conversion/MemRefToEmitC/MemRefToEmitC.cpp

```diff
@@ -177,7 +177,8 @@ void mlir::populateMemRefToEmitCTypeConversion(TypeConverter &typeConverter) {
         }
         Type convertedElementType =
             typeConverter.convertType(memRefType.getElementType());
-        if (!convertedElementType)
+        if (!convertedElementType ||
```

Comment:
I think we should instead change https://github.com/llvm/llvm-project/blob/main/mlir/lib/Conversion/MemRefToEmitC/MemRefToEmitCPass.cpp#L36 to 
```
    converter.addConversion([](Type type) -> std::optional<Type> {
      if (emitc::isSupportedEmitCType(type))
        return type;
      return {};
    });
```
which better allows to compose `populateMemRefToEmitCTypeConversion` with custom array element type conversions.

Reply:
Good suggestion! I agree with Matthias here, please move the check as proposed.

Reply:
Good idea. I'll change it right away.